pyriemann/utils/mean.py

```diff
@@ -48,20 +48,19 @@ def mean_ale(covmats, tol=10e-7, maxiter=50, sample_weight=None):
         <https://arxiv.org/abs/1505.07343>`_
         M. Congedo, B. Afsari, A. Barachant, M. Moakher. PLOS ONE, 2015
     """
-    n_matrices, n, _ = covmats.shape
+    if covmats is not None:
+        warnings.warn("Input covmats has been renamed into X and will be "
+                      "removed in 0.6.0.", category=DeprecationWarning)
```

Comment:
You miss here X=covmats

Reply:
Is this function ok for you?
```
def _deprecate_covmats(covmats, X):
    if covmats is not None:
        print("DeprecationWarning: input covmats has been renamed into X "
               "and will be removed in 0.6.0.")
        X = covmats
    return X
```

Reply:
yes and the signature should actually be 

mean_func(X=None, ..., covmats=None)

so doing

mean_func(covmats=covmats)

still works

Reply:
Ok. But `mean_power` and `maskedmean_riemann` can not support `X=None`.

Reply:
what do you mean?

Reply:
```
def maskedmean_riemann(X=None, masks, tol=10e-9, maxiter=100, init=None,
                       sample_weight=None, covmats=None):
```
raises `Non default argument follows defaut argument`. 
To solve that, I must define `X=None, masks=None` and raise an error if masks=None...

Same problem for
```
def mean_power(X, p, *, sample_weight=None, zeta=10e-10, maxiter=100,
               covmats=None):
```